### DISCUSSION TITLE
Expected behaviour of experimental_createPersister

### DISCUSSION THREAD

**Discussion by sgcullen** [TanStack/query#8466] [General] [answered]
Using the experimental_createPersister, i notice that if a `gcTime` and `maxAge` are set as follows as per the docs: https://tanstack.com/query/v5/docs/framework/react/plugins/createPersister

```
import AsyncStorage from '@react-native-async-storage/async-storage'
import { QueryClient } from '@tanstack/react-query'
import { experimental_createPersister } from '@tanstack/query-persist-client-core'

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      gcTime: 1000 * 30, // 30 seconds
      persister: experimental_createPersister({
        storage: AsyncStorage,
        maxAge: 1000 * 60 * 60 * 12, // 12 hours
      }),
    },
  },
})
```

...the data is stored to the persister storage but it is also refetched unless a `staleTime` is also set.

In the context of the experimental_createPersister, what is the difference between `maxAge` and `staleTime` and how best should they be used together?

Many thanks!

**Comment by TkDodo** [comment, marked as answer]
`maxAge` is the time until data should be used from the external storage. If we restore data and see it’s older than `maxAge`, we won’t use it.

`staleTime`, as usual, just defines if the data is fresh or stale. If it’s stale, we’ll give it to you and do a background refetch. If it’s fresh, we’ll give it to you without a refetch.
